### PR TITLE
Add tested and working version

### PR DESCRIPTION
**Pull request by Scriptkiddi**
Please ignore the sudos used. The npm module is not an option if the the
npm packages should not be installed as root.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# mattermost-hubot
+
+Install and configure hubot for the use with a Mattermost instance
```
Comment (SlothOfAnarchy) [inline README.md]: Installs*
dot at the end

File: README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+## Requirements
+
+A debian based system
```
Comment (SlothOfAnarchy) [inline README.md]: Debian-based

File: README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+| `mattermost_hubot_log_level`           | `info`                           | Set log level                                                                                     |
+| `mattermost_hubot_reply`               | `True`                           | Set to `False` to stop posting reply responses as comments                                        |
+| `mattermost_hubot_ignore_users`        | ` `                              | List of users that should be ignored                                                              |
+| `mattermost_hubot_owner`               | :heavy_check_mark:               | Mail of person that runs this bot                                                                 |
+| `mattermost_hubot_name`                | `James`                          | Name under which the bot should react                                                             |
+| `mattermost_hubot_description`         | `Your friendly and helpfull bot` | Description which the bot should use                                                              |
```
Comment (SlothOfAnarchy) [inline README.md]: *helpful
Also fix it in the defaults

File: README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+| `mattermost_hubot_enviroment`          | ` `                              | Directory to set more enviroment variables. The key ist the variable name and the value the value |
```
Comment (SlothOfAnarchy) [inline README.md]: *is

File: README.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+i
```
Comment (SlothOfAnarchy) [inline README.md]: i?

File: README.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  become: true
+  vars:
+    mattermost_hubot_mattermost_team: teamname
+    mattermost_hubot_mattermost_user: hubot
+    mattermost_hubot_mattermost_password: yourpassword
+    mattermost_hubot_mattermost_host: url.to.your.mattermost.instance.com
+    mattermost_hubot_owner: your@mailaddress.com
+    mattermost_hubot_enviroment:
+      - REDIS_URL: redis://:password@192.168.0.1:16379/prefix 
+    mattermost_hubot_plugins:
```
Comment (SlothOfAnarchy) [inline README.md:59]: not documented

File: handlers/main.yml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+---
+- name: Restart systemd
```
Comment (SlothOfAnarchy) [inline handlers/main.yml]: Restart mattermost-hubot service
(I think systemd per se will still run after this task.)

File: meta/main.yml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+      versions:
+        - xenial
+        - yakkety
+        - zesty
+        - artful
+
+  galaxy_tags:
+    - hubot
+    - mattermost
+    - personal-assisten
```
Comment (SlothOfAnarchy) [inline meta/main.yml]: assistant?

File: tasks/main.yml [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+  with_items:
+    - "{{ mattermost_hubot_install_path }}"
+    - "{{ mattermost_hubot_install_path }}/.npm"
+
+# We are using shell and sudo since npm does not respect the user under which a process should run
+- name: Install "yo" and "generator-hubot" node.js package
+  environment:
+    HOME: "{{ mattermost_hubot_install_path }}"
+    npm_config_user: "{{ mattermost_hubot_user }}"
+    npm_config_init_module: "{{ mattermost_hubot_install_path }}/.npm-init.js"
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: shouldn't these file exist before executing this? I don't see them being created.

File: tasks/main.yml [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+    npm_config_userconfig: "{{ mattermost_hubot_install_path }}/.npmrc"
+    npm_config_cache: "{{ mattermost_hubot_install_path }}/.npm"
+  shell: "sudo -E -u {{ mattermost_hubot_user }} npm install {{ item }}"
+  changed_when: False
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: why? I'm sure npm has a shell option that can be parsed for changes

File: tasks/main.yml [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+  args:
+    chdir: "{{ mattermost_hubot_install_path }}"
+  with_items:
+    - yo
+    - generator-hubot
+
+- name: Init hubot
+  become: True
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: Isn't this the default?

File: tasks/main.yml [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+  shell: "sudo -u {{ mattermost_hubot_user }} -E yo hubot --adapter matteruser --owner {{ mattermost_hubot_owner }} --name {{ mattermost_hubot_name }} --description '{{ mattermost_hubot_description }}'"
+  environment:
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: Can you extract the environment variables to vars and only define them once?

File: tasks/main.yml [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+  template:
+    src: systemd.j2
+    dest: /etc/systemd/system/mattermost-hubot.service
+
+- name: Place enviroment file
+  template:
+    src: hubot_enviroment
+    dest: "/etc/default/hubot"
+    mode: 0600
+    owner: "root"
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: why the quotes? same with `dest`

File: tasks/main.yml [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+    HOME: "{{ mattermost_hubot_install_path }}"
+    npm_config_user: "{{ mattermost_hubot_user }}"
+    npm_config_init_module: "{{ mattermost_hubot_install_path }}/.npm-init.js"
+    npm_config_userconfig: "{{ mattermost_hubot_install_path }}/.npmrc"
+    npm_config_cache: "{{ mattermost_hubot_install_path }}/.npm"
+  args:
+    chdir: "{{ mattermost_hubot_install_path }}"
+    creates: "{{ mattermost_hubot_install_path }}/scripts"
+
+- name: Delete hubot-scripts.json
```
Comment (SlothOfAnarchy) [inline tasks/main.yml:39]: why?

File: tasks/main.yml [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+    dest: "{{ mattermost_hubot_install_path }}/external-scripts.json"
+  notify:
+    - Restart systemd
+
+- name: Enable and start the systemd service
+  systemd:
+    name: mattermost-hubot
+    state: started
+    enabled: True
+    daemon_reload: True
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: Only reload daemon when the config changes.

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: tasks/main.yml (hunk 14 above)
Comment (Scriptkiddi) [reply tasks/main.yml:39]: because its deprecated and unneeded

File: tasks/main.yml (hunk 9 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: they dont need to exist

File: README.md (hunk 1 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 4 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 5 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 6 above)
Comment (Scriptkiddi) [reply README.md:59]: fixed

File: handlers/main.yml (hunk 7 above)
Comment (Scriptkiddi) [reply handlers/main.yml]: fixed

File: meta/main.yml (hunk 8 above)
Comment (Scriptkiddi) [reply meta/main.yml]: fixed

File: tasks/main.yml (hunk 10 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: there is a json option but it does not tell you what npm did only the deps of the installed packages

File: tasks/main.yml (hunk 11 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: tasks/main.yml (hunk 13 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: tasks/main.yml (hunk 15 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: tasks/main.yml (hunk 12 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: README.md [hunk 16]
```diff
@@ -1,34 +1,47 @@
...
+| `mattermost_hubot_ignore_users`        | ` `                             | List of users that should be ignored                                                                      |
+| `mattermost_hubot_owner`               | :heavy_check_mark:              | Mail of person that runs this bot                                                                         |
+| `mattermost_hubot_name`                | `James`                         | Name under which the bot should react                                                                     |
+| `mattermost_hubot_description`         | `Your friendly and helpful bot` | Description which the bot should use                                                                      |
+| `mattermost_hubot_enviroment`          | ` `                             | Directory to set more enviroment variables. The key is the variable name and the value the value          |
+| `mattermost_hubot_plugins`             | See below                       | List of plugins to be installed. Name of the package is the same name used in the npm package repository. |
```
Comment (SlothOfAnarchy) [inline README.md:31]: Maybe mention that it's a list but I would approve it without it.

File: handlers/main.yml [hunk 17]
```diff
@@ -1,5 +1,12 @@
 ---
-- name: Restart systemd
+- name: Restart mattermost-hubot service
   service:
     name: mattermost-hubot
     state: restarted
+
+- name: Reload systemd daemon and enable mattermost-hubot service
```
Comment (SlothOfAnarchy) [inline handlers/main.yml]: It is also being started.

**Review by SlothOfAnarchy — DISMISSED** [review]
see my comments if you want to fix them

**Comment by Scriptkiddi** [comment]
@SlothOfAnarchy please rereview

**Review by SlothOfAnarchy — APPROVED** [review]

**Pull request merged**